### DISCUSSION TITLE
javax.persistence.validation.mode=NONE not work [reactive]

### DISCUSSION THREAD

**Discussion by uPagge** [quarkusio/quarkus#22129] [Q&A]
Hello
i can't figure out how to set javax.persistence.validation.mode on reactive hibernate?
i tried to use persistence.xml but it cannot be used with reactive approach?
How do I turn off database-level validation?

**Comment by Raphasha27** [comment]
Set `javax.persistence.validation.mode` in `application.properties` — Quarkus doesn't read system properties for Hibernate settings:

```properties
# application.properties — this works
quarkus.hibernate-orm.validation.enabled=false
# OR
quarkus.hibernate-orm."javax.persistence.validation.mode"=none

# These do NOT work in Quarkus:
# javax.persistence.validation.mode=none  ← wrong, system property not config
```

For reactive Hibernate:

```properties
quarkus.hibernate-orm.reactive.validation.enabled=false
```

If you need to disable validation for specific entities:

```java
@Entity
@Table(name = "users")
@javax.persistence.PersistenceValidation(mode = ValidationMode.NONE)
public class User {
    // fields
}
```

To verify validation is off, check the logs for:

```
Hibernate: HV000041: No validator implementation found, validation will be skipped
```

If you still see validation errors, the problem might be:

1. **Bean Validation dependency** — remove it if not needed:
```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-hibernate-validator</artifactId>
    <scope>test</scope>  <!-- or remove entirely -->
</dependency>
```

2. **RESTEasy validation** — separate from Hibernate validation. Disable at REST level:
```properties
quarkus.rest-reactive.validation.enabled=false
```

3. **`@Valid` annotations** on REST endpoints trigger validation regardless of Hibernate settings. Remove `@Valid` from endpoint parameters if you want no validation.
